types/setting.go

```diff
@@ -1942,116 +1921,389 @@ func GetDangerZoneSettings() sets.Set[SettingName] {
 	return settingList
 }
 
-func validateBool(definition SettingDefinition, value string) error {
-	if definition.Type != SettingTypeBool {
-		return nil
+// IsJSONFormat checks if the input string starts with '{' indicating JSON format.
+func IsJSONFormat(value string) bool {
+	return strings.HasPrefix(strings.TrimSpace(value), "{")
+}
+
+// validateBool validates a boolean setting value based on the provided definition.
+// It supports both single boolean values and JSON-formatted data-engine-specific values.
+func validateBool(definition SettingDefinition, value string) (err error) {
+	if IsJSONFormat(strings.TrimSpace(value)) {
+		_, err = ParseBoolsInJSONFormat(definition, value)
+	} else {
+		_, err = parseSingleBool(definition, value)
 	}
 
-	if value != "true" && value != "false" {
-		return fmt.Errorf("value %v should be true or false", value)
+	return err
+}
+
+// ParseBoolsInJSONFormat processes JSON-formatted input for data-engine-specific boolean settings.
+func ParseBoolsInJSONFormat(definition SettingDefinition, value string) (map[longhorn.DataEngineType]any, error) {
```

Comment:
The function name feels a bit too generic. Consider renaming it to reflect that it’s parsing data-engine-specific values. The same applies to the other new functions as well.

Reply:
Good idea. Updated to `ParseDataEngineSpecificSettingStrings{Ints,Bools, ...}`. Thank you.